index.bs

```diff
@@ -1616,27 +1673,37 @@ partial interface MLGraphBuilder {
     **Returns:** an {{MLOperand}}. The permuted or transposed N-D tensor. 
 </div>
 
-## MLGraph ## {#api-mlgraph}
-The {{MLGraph}} interface represents a compiled computational graph. A compiled graph once constructed is immutable and cannot be subsequently changed.
+## MLTensor ## {#api-mltensor}
+The {{MLTensor}} interface represents a resource of a multidimensional array of numbers with a shape.
 
 <script type=idl>
-dictionary MLInput {
-  required (MLBufferView or WebGLTexture or GPUTexture) data;
-  sequence<long> dimensions;
-};
+typedef (MLBufferView or WebGLTexture or GPUTexture) MLResource;
 
-dictionary MLOutput {
-  (MLBufferView or WebGLTexture or GPUTexture) data;
-  sequence<long> dimensions;
+[SecureContext, Exposed=Window]
+interface MLTensor {
+  // Construct a tensor from a resource
+  constructor(MLResource resource, optional sequence<long> dimensions);
+
+  // The tensor dimensions
+  sequence<long> dimensions();
+
+  // The underlying resource of the tensor
+  MLResource resource();
+
+  // Asynchronously download the result of a computation onto a typed array.
+  Promise<ArrayBufferView> data();
```

Comment:
I would propose to support downloading the result into a pre-allocated `ArrayBufferView`. e.g.,
```js
Promise<(ArrayBufferView or undefined)> data(optional ArrayBufferView buffer);
```

If `buffer` is not provided, the returned promise resolves with a newly created `ArrayBufferView` contains the downloaded result.
If `bufer` is provided, the returned promise resolves with `undefined` after downloading the result into user provided `buffer`.

This would allow Wasm libs download the data to the existing Wasm memory instead of making another copy, as @pyu10055 mentioned in the [CPU - WebAssembly example](https://github.com/webmachinelearning/proposals/issues/2#issuecomment-773441501) of operation-level execution use case.